src/zeep/transports.py

```diff
@@ -24,17 +27,18 @@ def __init__(self, cache=NotSet, timeout=300, operation_timeout=None,
         :param verify: Boolean to indicate if the SSL certificate needs to be
                        verified.
         :param http_auth: HTTP authentication, passed to requests.
+        :param http_headers: Custom HTTP headers dict, passed to requests.
 
         """
-        self.cache = SqliteCache() if cache is NotSet else cache
+        self.cache = cache
         self.load_timeout = timeout
         self.operation_timeout = operation_timeout
         self.logger = logging.getLogger(__name__)
 
         self.http_verify = verify
         self.http_auth = http_auth
-        self.http_headers = {
-            'User-Agent': 'Zeep/%s (www.python-zeep.org)' % (get_version())
+        self.http_headers = http_headers or {
+            'User-Agent': ZEEP_USERAGENT,
         }
```

Comment:
This isn't the same. In this case the user agent isn't set if someone passes custom headers